packages/block-editor/src/store/reducer.js

```diff
@@ -1650,6 +1651,18 @@ export function highlightedBlock( state, action ) {
 	return state;
 }
 
+export function hoveredBlocks( state = [], action ) {
+	switch ( action.type ) {
+		case 'HOVERED_BLOCKS_ADD':
+			return [ action.clientId, ...state ];
+
+		case 'HOVERED_BLOCKS_REMOVE':
+			return filter( state, ( blockId ) => blockId !== action.clientId );
```

Comment:
some ideas:
- if we need to keep track of multiple, what about using a plain object (e.g. mapping `clientId => true`)?
- is the reason why we include multiple blocks (instead of tracking a single clientId) to make sure that parent blocks are also considered "hovered" when the children are also hovered?

Reply:
> is the reason why we include multiple blocks (instead of tracking a single clientId) to make sure that parent blocks are also considered "hovered" when the children are also hovered?

Partially, yeah.  I haven't figured out a good way to track 1 block.  Basically nested blocks create the issue, if we are setting the id onMouseEnter and removing it onMouseLeave, then mousing into nested blocks would set blocks appropriately, but mousing out of nested blocks would clear the state entirely when things are still actually hovered.

Also, if we tracked a single clientId, we would need to bubble up through its parents to see if any are a template part.  However, getBlockParents returns an empty array for blocks at the top level inside a template part, and they don't actually return the template part block as a parent.  So instead we just keep track of all of the blocks that are currently hovered and go through that list to see if any match what we are looking for.

> if we need to keep track of multiple, what about using a plain object (e.g. mapping clientId => true)?
Yeah possibly.  there could still be some benefit to keeping the order they are added, but that could be handled with an array in that object as well or something.

Reply:
`getBlockParents` ah.... that is probably "unexpected" behavior around the controlled inner blocks code. Well. Kind of expected, but not really. The idea is that the "template part block" is not actually the parent of those blocks _outside_ of the block-editor (e.g. from the point of view of the template part entity). So that's why some of this stuff doesn't work as expected.